### ISSUE TITLE
Extra newlines are lost switching to/from blockMode

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/codemirror-blocks#191]
```
; an important constant!
(define x 5)


; some code
(+ x 2)
```

becomes:

```
#| an important constant! |#
(define x 5)
#| some code |#
(+ x 2)
```

Shriram points out that curriculum designers use extra newlines all the time, to space things out. I know we can't preserve this if roots are re-ordered, but is there ANY ability for us to keep existing newlines in the current pretty-printer?

**Comment by justinpombrio** [comment]
Yeah, this has irked me too. It shouldn't be too difficult to preserve toplevel newlines. Newlines inside of blocks (e.g. inside of a long function definition) are much harder and come with lots of corner cases.

**Comment by schanzer** [comment]
Newlines inside roots are way, WAY less important. If we can solve the between-roots case, we'll be in good shape.

**Comment by justinpombrio** [comment]
Fixed by ce09fcbfb829882605c2147bc5bc2779dc06ea02.

(Two notes. First, this commit always puts at least one newline (newline, not blank line) between blocks. I believe this is what we want; if not it's easy to change. Second, it only preserves toplevel blank lines. If you want to preserve more than that, open a new issue and we can discuss all the awful corner cases I suspect this brings up.)
